bin/apisix

```diff
@@ -540,24 +540,22 @@ local function check_or_version(cur_ver_s, need_ver_s)
     return true
 end
 
+local function system_dns_resolver(file_path)
+    local file, err = io.open(file_path, "rb")
+    if not file then
+        return false, "failed to open file: " .. file_path .. ", error info:" .. err
+    end
 
-local function system_dns_resolver()
-    local file, err = io.open("/etc/resolv.conf")
-    if err then
-        return nil, err
-    else
-        local dns_addrs = {}
-        for line in file:lines() do
-            local addr, n = line:gsub("^nameserver%s+(%d+%.%d+%.%d+%.%d+)%s*$", "%1")
-            if n == 1 then
-                table.insert(dns_addrs, addr)
-            end
+    local dns_addrs = {}
+    for line in file:lines() do
+        local addr, n = line:gsub("^nameserver%s+(%d+%.%d+%.%d+%.%d+)%s*$", "%1")
+        if n == 1 then
+            table.insert(dns_addrs, addr)
         end
-        return dns_addrs
     end
+    return dns_addrs
```

Comment:
We better close the file handle before exiting the function. `file:close()`

Reply:
the old codes not close file too.
@membphis Do we have test cases to cover `bin/apisix`? only `apisix start/stop` looks not good enough.

Reply:
Oops missed that @membphis  fixed it  :smiley: